packages/@react-aria/toast/src/useToast.ts

```diff
@@ -58,44 +61,35 @@ export function useToast<T>(props: AriaToastProps<T>, state: ToastState<T>, ref:
     };
   }, [timer, timeout]);
 
-  // Restore focus to the toast container on unmount.
-  // If there are no more toasts, the container will be unmounted
-  // and will restore focus to wherever focus was before the user
-  // focused the toast region.
-  let focusOnUnmount = useRef(null);
-  useLayoutEffect(() => {
-    let container = ref.current.closest('[role=region]') as HTMLElement;
-    return () => {
-      if (container && container.contains(document.activeElement)) {
-        // Focus must be delayed for focus ring to appear, but we can't wait
-        // until useEffect cleanup to check if focus was inside the container.
-        focusOnUnmount.current = container;
-      }
-    };
-  }, [ref]);
-
-  // eslint-disable-next-line
+  let [isEntered, setIsEntered] = React.useState(false);
   useEffect(() => {
-    return () => {
-      if (focusOnUnmount.current) {
-        focusOnUnmount.current.focus();
-      }
-    };
-  }, [ref]);
+    if (animation === 'entering') {
+      setIsEntered(true);
+    }
+  }, [animation]);
 
   let titleId = useId();
   let descriptionId = useSlotId();
   let stringFormatter = useLocalizedStringFormatter(intlMessages, '@react-aria/toast');
 
   return {
     toastProps: {
-      role: 'alert',
+      role: 'alertdialog',
+      'aria-modal': 'false',
```

Comment:
@majornista correct me if I'm wrong, but this is just to possibly forward proof? since alertdialog would be considered a modal by default, and so we use aria-modal to communicate that it's not, in fact, a modal

Reply:
`aria-modal="false"` doesn't really make a difference for screen reader behavior; the default is for a dialog to be non-modal. However, the [APG design pattern](https://www.w3.org/WAI/ARIA/apg/patterns/alertdialog/) describes an `alertdialog` as modal, so I thought it best to be explicit:

> An alert dialog is a [modal dialog](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/) that interrupts the user's workflow to communicate an important message and acquire a response. Examples include action confirmation prompts and error message confirmations. The [alertdialog](https://w3c.github.io/aria/#alertdialog) role enables assistive technologies and browsers to distinguish alert dialogs from other dialogs so they have the option of giving alert dialogs special treatment, such as playing a system alert sound.

I use `role="alertdialog"`, as a container for the toast message content, which has `role="alert"` to announce as a live region, and the button to dismiss the Toast, for a few reasons.

1. A Toast is kind of a tiny non-modal dialog, so it didn't seem like much of a stretch.
2. Using a dialog as a container helps disambiguate between the Dismiss buttons when more than one Toast is displayed.
3. When we close a Toast, unlike the Dismiss button for an adjacent Toast, the dialog provides a uniquely labeled widget element to receive focus, and is easy to style using the React-Spectrum focus ring.
4. Depending on the assistive technology being used, an alert dialog will open with an earcon, which I see as nice feature to distinguish a Toast from other content the screen reader may be announcing.